elements/pfe-absolute-position/src/pfe-absolute-position.js

```diff
@@ -0,0 +1,421 @@
+        type: Boolean,
+        default: false
+      },
+      /**
+       * If true, no parts of the tooltip will ever be shown offscreen.
+       */
+      hidden: {
+        type: Boolean,
+        reflect: true
+      },
+      /**
+       * The id of the element that the tooltip is anchored to. This element
+       * must be a sibling of the tooltip. If this property is not set,
+       * then the tooltip will be centered to the parent node containing it.
+       */
+      for: {
+        type: String,
+        reflect: true,
+        default: null
+      },
+      /**
+       * The spacing between the top of the tooltip and the element it is
+       * anchored to.
+       */
+      offset: {
+        type: Number,
+        default: 0
+      },
+      /**
+       * Positions the tooltip to the top, right, bottom, left of its content.
+       */
+      position: {
+        type: String,
+        reflect: true,
+        default: "top"
+      },
+      /**
+       * Aligns at the start, or end fo target. Default is centered.
+       */
+      positionAlign: {
+        type: String,
+        reflect: true,
+        default: "center"
+      },
+      /**
+       * Set the mode to either hover or click
+       */
+      mode: {
+        type: String,
+        default: "hover"
+      }
+    };
+  }
+
+  /**
+   * Get the actual target element
+   * @type {Node}
+   */
+  get target() {
+    return this._target;
+  }
+
+  /**
+   * Set the actual target element
+   * @type {Node}
+   */
+  set target(node) {
+    this._target = node;
+    return this._target;
+  }
+
+  /**
+   * Get the actual target element
+   * @type {Object}
+   */
+  get _positions() {
+    return this._target;
+  }
+
+  /**
+   * Set the actual target element
+   * @type {Object}
+   */
+  set _positions(positions) {
+    this.__positions = positions;
+    return this.__positions;
+  }
+
+  // Declare the type of this component
+  static get PfeType() {
+    return PFElement.PfeTypes.Container;
+  }
+
+  constructor() {
+    super(PfeAbsolutePosition, { type: PfeAbsolutePosition.PfeType });
+    this._positions = {};
+    this.target = null;
+    this.__observe = false;
+    this.__manager = window.AbsolutePositionStateManager.requestAvailability();
+
+    this.animationEntry = "";
+    this.animationExit = "";
+    this.animationConfig = {
+      entry: [{ name: "fade-in-animation", node: this, timing: { delay: 0 } }],
+      exit: [{ name: "fade-out-animation", node: this }]
+    };
+    setTimeout(() => {
+      this.addEventListener("webkitAnimationEnd", this._onAnimationEnd.bind(this));
+    }, 0);
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    Object.keys(PfeAbsolutePosition.properties).forEach(prop => this._updated(prop, null, this[prop]));
+  }
+
+  attributeChangedCallback(attr, oldVal, newVal) {
+    super.attributeChangedCallback(...arguments);
+    this._updated(...arguments);
+  }
+
+  _updated(attr, oldVal, newVal) {
```

Comment:
I'd recommend splitting this out into separate observers and adding them to the properties object above.  Then you're only running the relevant code and you're not having to extend attributeChangedCallback for every attribute.

Reply:
Yeah absolutely!

Reply:
Moved to property observers 👍